akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/Effect.scala

```diff
@@ -203,6 +203,20 @@ object Effect {
     def duration(): java.time.Duration = delay.asJava
   }
 
+  final case class TimerScheduled[U](key: Any, msg: U, delay: FiniteDuration, mode: TimerScheduled.TimerMode)
+      extends Effect {
+    def duration(): java.time.Duration = delay.asJava
+  }
+
+  object TimerScheduled {
+    sealed trait TimerMode
+    case object FixedRateMode extends TimerMode
+    case object FixedDelayMode extends TimerMode
+    case object SingleMode extends TimerMode
+  }
```

Comment:
Check that these are available from Java as well without some Scala `MODULE$` in the object name and if not create singleton-accessor methods with the same names for them (`def fixedRateMode()`)

Reply:
seems like Java can't cope with classes declared inside a two levels deep nested object, do you have a suggestion or a pointer to a similar declaration that was exposed to Java?